### PR TITLE
minimize compile dependencies to kafka-clients only

### PR DESCRIPTION
**Pull request by radai-rosenblatt**
no more exporting scala onto innocent users.
TBD: rewrite test harnesses to get rid of scala entirely
Signed-off-by: rrosenbl <rrosenblatt@linkedin.com>

### CODE REVIEW

**Comment by benmccann** [comment]
Should we also add a couple lines to export a test jar? I think we have other code that needs the test harnesses.  http://mrhaki.blogspot.com/2010/11/gradle-goodness-create-jar-artifact.html

Thanks for doing this! This will be an enormous help!!

**Comment by radai-rosenblatt** [comment]
i seriously dislike the current test harness code.
I'd like to convert it to java and move it to be its own module (test-support? test-harness?). 
that way there'd be no need to produce a test artifact (as a test artifact would include our actual tests, not just the harness)

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,10 +1,6 @@
...
 See License in the project root for license information.
-#
-
-#Fri Apr 22 16:57:09 PDT 2016
+#Mon Feb 13 17:24:15 PST 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.2-bin.zip
```
Comment (benmccann) [inline gradle/wrapper/gradle-wrapper.properties]: do you really have to change this line?

Comment (radai-rosenblatt) [inline gradle/wrapper/gradle-wrapper.properties]: the switch from *-bin to *-all is meant to enable auto-complete in IDEA for gradle build files. the artifact is larger, but its a one-time setup cost and I think the end result is worth it

Comment (benmccann) [inline gradle/wrapper/gradle-wrapper.properties]: Sounds good then. Thanks for the clarification

**Pull request merged**